include/glaze/json/read.hpp

```diff
@@ -1028,30 +1028,33 @@ namespace glz
             }
 
             auto start = it;
+            skip_string_view<Opts>(ctx, it, end);
+            if (bool(ctx.error)) [[unlikely]]
+               return;
 
             if constexpr (string_view_t<T>) {
-               skip_string_view<Opts>(ctx, it, end);
-               if (bool(ctx.error)) [[unlikely]]
-                  return;
                value = {start, size_t(it - start)};
-               ++it; // skip closing quote
-               GLZ_VALID_END();
             }
             else if constexpr (char_array_t<T>) {
-               skip_string_view<Opts>(ctx, it, end);
-               if (bool(ctx.error)) [[unlikely]]
-                  return;
-
                const size_t n = it - start;
-               if ((sizeof(value) - 1) < n) {
+               if ((n + 1) > sizeof(value)) {
                   ctx.error = error_code::unexpected_end;
                   return;
                }
                std::memcpy(value, start, n);
                value[n] = '\0';
-               ++it; // skip closing quote
-               GLZ_VALID_END();
             }
+            else if constexpr (array_char_t<T>) {
+               const size_t n = it - start;
+               if ((n + 1) > value.size()) {
+                  ctx.error = error_code::unexpected_end;
+                  return;
+               }
+               std::memcpy(value.data(), start, n);
+               value[n] = '\0';
```

Comment:
I would actually not expect that. For me, it should behave like a string of fixed-length without the null terminating char. If I have a `std::array<char, N>`, I would expect a string of exactly N chars. This follows-up my previous comment about how I would see the semantics of the std::array<char, N> parsing case.